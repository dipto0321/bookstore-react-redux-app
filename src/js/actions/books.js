import uuid from 'uuid';
import {
  ADD_BOOK,
  REMOVE_BOOK,
  SWITCH_ADD_BOOK_MODE,
} from './actionTypes';


const addBook = ({ title, author, category }) => ({
  type: ADD_BOOK,
  book: {
    id: uuid(),
    title,
    author,
    category,
  },
});

const switchAddBookMode = () => ({
  type: SWITCH_ADD_BOOK_MODE,
});

const removeBook = id => ({
  type: REMOVE_BOOK,
  id,
});

export {
  addBook,
  removeBook,
  switchAddBookMode,
};
