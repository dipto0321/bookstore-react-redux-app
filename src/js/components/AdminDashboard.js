import React from 'react';
import {
  Tab, Tabs, TabList, TabPanel,
} from 'react-tabs';
import 'react-tabs/style/react-tabs.scss';
import SearchForm from './SearchForm';

const AdminDashboard = () => (
  <Tabs>
    <TabList>
      <Tab>Manage Users</Tab>
      <Tab>Manage Books</Tab>
    </TabList>
    <TabPanel>
      <h2>Users</h2>
    </TabPanel>
    <TabPanel>
      <SearchForm />
    </TabPanel>
  </Tabs>
);

export default AdminDashboard;
