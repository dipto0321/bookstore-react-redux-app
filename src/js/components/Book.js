import React from 'react';
import PropTypes from 'prop-types';

const Book = (props) => {
  const { handleRemove, book } = props;
  const { id, title, category } = book;

  return (
    <div className="card">
      <div className="card-body">
        <div className="columns">
          <div className="col">
            <h4 className="genre">{category}</h4>
            <h3 className="title">{title}</h3>
            <div className="action">
              <button type="button" onClick={() => handleRemove(id)}>
                Remove
              </button>
            </div>
          </div>
          <div className="col" />
        </div>
      </div>
    </div>
  );
};

Book.propTypes = {
  book: PropTypes.instanceOf(Object).isRequired,
  handleRemove: PropTypes.func.isRequired,
};

export default Book;
