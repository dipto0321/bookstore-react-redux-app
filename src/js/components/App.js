import React from 'react';
import { Route, Router, Switch } from 'react-router-dom';
import { Provider } from 'react-redux';
import { createBrowserHistory } from 'history';
import decode from 'jwt-decode';
import BooksDashboard from './BooksDashboard';
import '../../scss/main.scss';
import LoginPage from '../containers/LoginPage';
import configureStore from '../store/configureStore';
import { setCurrentUser } from '../actions/currentUser';
import { setAuthorizationToken } from '../services/api';
import withAuth from '../hocs/withAuth';

export const history = createBrowserHistory();

const store = configureStore();

if (localStorage.token) {
  const userData = decode(localStorage.token);
  setAuthorizationToken(localStorage.token);

  try {
    store.dispatch(setCurrentUser({ authenticated: true, data: userData }));
  } catch (err) {
    store.dispatch(setCurrentUser({ authenticated: false, data: null }));
  }
}

const App = () => (
  <Provider store={store}>
    <Router history={history}>
      <Switch>
        <Route path="/login" component={LoginPage} />
        <Route path="/" component={withAuth(BooksDashboard, store.getState().currentUser.authenticated)} />
      </Switch>
    </Router>
  </Provider>
);

export default App;
