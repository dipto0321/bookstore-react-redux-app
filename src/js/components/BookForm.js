import React, { useState } from 'react';
import { connect } from 'react-redux';
import { addBookToLibrary } from '../thunks/book';
import { switchAddBookMode } from '../actions/book';
import InputWrapper from './InputWrapper';

export const BookForm = ({ book, addBookToLibrary, switchAddBookMode }) => {
  const [title, setTitle] = useState(book ? book.title : '');
  const [subTitle, setSubTitle] = useState(book ? book.subtitle : '');
  const [authors, setAuthors] = useState(book ? book.authors.join(', ') : '');
  const [description, setDescription] = useState(book ? book.description : '');
  const [publishedDate, setPublishedDate] = useState(book ? book.publishedDate : '');
  const [thumbNail, setThumbNail] = useState(book ? book.imageLinks.smallThumbnail : '');
  const [isbn, setIsbn] = useState(book ? book.industryIdentifiers[0].identifier : '');
  const [pageCount, setPageCount] = useState(book ? book.pageCount : '');
  const [category, setCategory] = useState(book ? book.categories[0] : '');

  const handleSubmit = (e) => {
    e.preventDefault();
    addBookToLibrary({
      book: {
        title,
        subtitle: subTitle,
        authors,
        description,
        published_date: publishedDate,
        thumbnail: thumbNail,
        isbn,
        page_count: pageCount,
        category,
      },
    })
      .then(() => switchAddBookMode());
  };

  return (
    <form>
      <InputWrapper
        inputValue={title}
        labelValue="Title: "
        setInput={setTitle}
        type="text"
        inputId="title"
        error={null}
      />
      <InputWrapper
        inputValue={subTitle}
        labelValue="Sub-title: "
        setInput={setSubTitle}
        type="text"
        inputId="sub-title"
        error={null}
      />
      <InputWrapper
        inputValue={authors}
        labelValue="Authors: "
        setInput={setAuthors}
        type="text"
        inputId="authors"
        error={null}
      />
      <InputWrapper
        inputValue={category}
        labelValue="Category: "
        setInput={setCategory}
        type="text"
        inputId="category"
        error={null}
      />
      <InputWrapper
        inputValue={isbn}
        labelValue="ISBN: "
        setInput={setIsbn}
        type="text"
        inputId="isbn"
        error={null}
      />
      <InputWrapper
        inputValue={description}
        labelValue="Description: "
        setInput={setDescription}
        type="textarea"
        inputId="description"
        error={null}
      />
      <InputWrapper
        inputValue={publishedDate}
        labelValue="Published Date: "
        setInput={setPublishedDate}
        type="text"
        inputId="published-date"
        error={null}
      />
      <InputWrapper
        inputValue={pageCount}
        labelValue="Page Count: "
        setInput={setPageCount}
        type="text"
        inputId="page-count"
        error={null}
      />
      <InputWrapper
        inputValue={thumbNail}
        labelValue="Thumbnail Link: "
        setInput={setThumbNail}
        type="text"
        inputId="thumbnail"
        error={null}
      />
      <div className="btn-wrapper">
        <button
          type="submit"
          className="add-book-btn"
          onClick={handleSubmit}
        >
          Add Book To Library
        </button>
      </div>
    </form>
  );
};

const mapStateToProps = state => ({
  book: state.addBookMode.book,
});

export default connect(mapStateToProps, { addBookToLibrary, switchAddBookMode })(BookForm);
