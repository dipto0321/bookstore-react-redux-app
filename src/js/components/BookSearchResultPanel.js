import React from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import { switchAddBookMode } from '../actions/book';

export const BookSearchResultPanel = ({
  book, switchAddBookMode,
}) => {
  const {
    id, title, subtitle, description, authors, imageLinks,
  } = book;
  const authorNames = authors.map(author => author).join(', ');
  return (
    <div className="book-result-item">
      <Link
        to={{
          pathname: `/admin/book-search/${id}`,
          book,
        }}
      >
        <h3>
          {title}
            :
          {' '}
          {subtitle}
        </h3>
      </Link>
      <div className="book-result-item__details">
        <div className="book-result-item__image-wrapper">
          <img src={imageLinks.smallThumbnail} alt={title} />
        </div>
        <div className="book-result-item__infos">
          <h4>
            By:
            {' '}
            {authorNames}
            {' '}
          </h4>
          <p>
            {description.substr(0, 500)}
            {' '}
            <button
              type="button"
              className="add-book-btn"
              onClick={switchAddBookMode}
            >
              Add To Library
            </button>
          </p>
        </div>
      </div>
    </div>
  );
};

export default connect(null, { switchAddBookMode })(BookSearchResultPanel);
