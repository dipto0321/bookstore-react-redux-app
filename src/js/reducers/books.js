import books from "../data/dummyBooks";

export default (state = books, action) => {
  switch (action.type) {
    case "ADD_BOOK":
      return [...state, action.book];
    case "REMOVE_BOOK":
      return state.filter(book => book.id !== action.id);
    default:
      return state;
  }
};
