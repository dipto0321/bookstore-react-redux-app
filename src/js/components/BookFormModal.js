import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import Modal from 'react-modal';
import BookForm from './BookForm';
import { switchAddBookMode } from '../actions/book';

export const BookFormModal = ({ switchAddBookMode, addBookMode }) => (
  <Modal
    isOpen={addBookMode}
    ariaHideApp={false}
    onRequestClose={switchAddBookMode}
    closeTimeoutMS={200}
  >
    <h3>Book Form Modal Here</h3>
  </Modal>
);

const mapStateToProps = state => ({
  addBookMode: state.addBookMode,
});

export default connect(mapStateToProps, { switchAddBookMode })(BookFormModal);
