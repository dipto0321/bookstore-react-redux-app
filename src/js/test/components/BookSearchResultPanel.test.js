import React from 'react';
import { shallow } from 'enzyme';
import { BookSearchResultPanel } from '../../components/BookSearchResultPanel';
import books from '../fixtures/books';

describe('BookSearchResultPanel', () => {
  const switchAddBookMode = jest.fn();
  const wrapper = shallow(
    <BookSearchResultPanel
      book={books[0]}
      switchAddBookMode={switchAddBookMode}
    />,
  );

  test('should render corectly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
