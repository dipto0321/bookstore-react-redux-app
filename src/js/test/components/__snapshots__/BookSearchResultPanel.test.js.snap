// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookSearchResultPanel should render corectly 1`] = `
<div
  className="book-result-item"
>
  <Link
    to={
      Object {
        "book": Object {
          "authors": Array [
            "Tolkien",
          ],
          "category": "Action",
          "chapters": "35",
          "currentChapter": "",
          "description": "Nostrud aliquip velit eu ea amet qui exercitation anim mollit do minim commodo pariatur.",
          "id": "0",
          "imageLinks": Object {
            "smallThumbnail": "",
          },
          "title": "Lord of the Rings",
        },
        "pathname": "/admin/book-search/0",
      }
    }
  >
    <h3>
      Lord of the Rings
      :
       
    </h3>
  </Link>
  <div
    className="book-result-item__details"
  >
    <div
      className="book-result-item__image-wrapper"
    >
      <img
        alt="Lord of the Rings"
        src=""
      />
    </div>
    <div
      className="book-result-item__infos"
    >
      <h4>
        By:
         
        Tolkien
         
      </h4>
      <p>
        Nostrud aliquip velit eu ea amet qui exercitation anim mollit do minim commodo pariatur.
         
        <button
          className="add-book-btn"
          type="button"
        >
          Add To Library
        </button>
      </p>
    </div>
  </div>
</div>
`;
